src/common.js

```diff
@@ -1,7 +1,74 @@
+var SHA256 = require('crypto-js/sha256');
 function Common() {}
 
 Common.prototype.exampleMethod = function () {
     return 'I am an example';
 }
 
-module.exports = Common;
\ No newline at end of file
+Common.prototype.logId5Id = function (id5Id) {
+    debugger;
+    //Checks and saves ID5 ID if it is new
+    if (id5Id != this.id5Id) {
+        this.id5Id = id5Id;
+        this.id5IdSent = false
+    }
+
+    //Sets user attribute if ID is unsent.
+    //This function will be updated once the decryption architecture is finalized.
+    //The ID may need to be sent as custom event.
+    if (this.id5IdSent == false){
+        var currentUser = mParticle.Identity.getCurrentUser();
+        currentUser.setUserAttribute('ID5ID', id5Id);
+        this.id5IdSent = true;
+    }
+}.bind(this);
+
+Common.prototype.buildPartnerData = function (mParticleUser) {
+    var pdKeys = {};
+    var userIdentities = mParticleUser.getUserIdentities();
+
+    var email = userIdentities.userIdentities['email'];
+    if (email) {
+        var processedEmail = SHA256(this.normalizeEmail(email));
+        pdKeys[1] = processedEmail;
+    }
```

Comment:
If this variable isn't being used anywhere else, I think you can just directly assign.

```suggestion
    if (email) {
        pdKeys[1] = SHA256(this.normalizeEmail(email));
    }
```

Reply:
updated